File: tests/fixtures/sample_dart/calculator.dart
/// A simple calculator that performs basic arithmetic operations.
///
/// This module provides basic arithmetic operations through the Calculator class.
class Calculator {
  /// The current value stored in the calculator.
  int _value;

  /// Initialize calculator with an optional starting value.
  ///
  /// [initialValue] is the starting value (default: 0).
  Calculator([int initialValue = 0]) : _value = initialValue;

  /// Add two numbers and return the result.
  ///
  /// [x] is the first number.
  /// [y] is the second number.
  /// Returns the sum of x and y.
  int add(int x, int y) {
    return x + y;
  }

  /// Multiply two numbers and return the result.
  ///
  /// [x] is the first number.
  /// [y] is the second number.
  /// Returns the product of x and y.
  int multiply(int x, int y) {
    return x * y;
  }

  /// Divide x by y and return the result.
  ///
  /// [x] is the numerator.
  /// [y] is the denominator.
  /// Returns the result of division.
  double divide(double x, double y) {
    return x / y;
  }

  /// Private method (should be marked as private in index).
  String _privateMethod() {
    return 'private';
  }

  /// Calculate a complex expression using multiple operations.
  ///
  /// [x] is the first operand.
  /// [y] is the second operand.
  /// [z] is the third operand.
  /// Returns the result of (x + y) * z.
  int calculateExpression(int x, int y, int z) {
    final sumResult = add(x, y);
    return multiply(sumResult, z);
  }
}

/// Top-level helper function to process data.
///
/// [data] is the list of items.
/// Returns the length of the list.
int helperFunction(List<int> data) {
  return data.length;
}

/// Private function (prefixed with underscore).
void _privateFunction() {
  // Private implementation
}

File: tests/fixtures/sample_ruby/calculator.rb
# frozen_string_literal: true

# A simple calculator that performs basic arithmetic operations.
#
# This module provides basic arithmetic operations through the Calculator class.
class Calculator
  # The current value stored in the calculator.
  attr_accessor :value

  # Initialize calculator with an optional starting value.
  #
  # @param initial_value [Integer] The starting value (default: 0)
  def initialize(initial_value = 0)
    @value = initial_value
  end

  # Add two numbers and return the result.
  #
  # @param x [Integer] First number
  # @param y [Integer] Second number
  # @return [Integer] Sum of x and y
  def add(x, y)
    x + y
  end

  # Multiply two numbers and return the result.
  #
  # @param x [Integer] First number
  # @param y [Integer] Second number
  # @return [Integer] Product of x and y
  def multiply(x, y)
    x * y
  end

  # Divide x by y and return the result.
  #
  # @param x [Numeric] Numerator
  # @param y [Numeric] Denominator
  # @return [Float] Result of division
  def divide(x, y)
    x.to_f / y
  end

  # Calculate a complex expression using multiple operations.
  #
  # @param x [Integer] First operand
  # @param y [Integer] Second operand
  # @param z [Integer] Third operand
  # @return [Integer] Result of (x + y) * z
  def calculate_expression(x, y, z)
    sum_result = add(x, y)
    multiply(sum_result, z)
  end

  private

  # Private method (should be marked as private in index).
  #
  # @return [String] A private string
  def private_method
    'private'
  end
end

# Top-level helper function to process data.
#
# @param data [Array] Array of items
# @return [Integer] Length of the array
def helper_function(data)
  data.length
end

# Private function at module level.
def private_function
  # Private implementation
end

private :private_function
